topics/single-cell/tutorials/pseudobulk-analysis/tutorial.md

```diff
@@ -0,0 +1,523 @@
+>    {: .comment}
+>
+{: .hands_on}
+
+> <question-title> What are the outputs of the Decoupler tool? </question-title>
+>
+> 1. How many outputs does the Decoupler tool generate?
+> 2. What do the outputs represent? What is the interpretation of the plots?
+> 3. Which output(s) will we use in edgeR for differential expression analysis?
+>
+{: .question}
+
+<solution-title>Solutions</solution-title>
+
+> 1. The Decoupler tool generates multiple outputs, including:
+>    - **Pseudobulk Count Matrix** (tabular file)
+>    - **Samples Metadata (Factor file)** (tabular file)
+>    - **Genes Metadata** (tabular file)
+>    - **Pseudobulk Plot** (PNG format)
+>    - **Filter by Expression Plot** (PNG format)
+>    - **Genes to Ignore by Contrast Field** (tabular file)
+>    - **Pseudobulk AnnData file** (If chosen, h5ad file) 
+>
+> 2. The output files contain the following:
+>    - **Pseudobulk Count Matrix:** Contains the raw count aggregates for each pseudobulk sample. 
+>    - **Samples Metadata (Factor File):** Provides metadata annotated for each sample, including factors or annotations added to the AnnData object.
+>    - **Genes Metadata:** Includes gene-related information such as gene symbols, Ensembl IDs, dispersion values, etc.
+>    - **Genes to Ignore:** Lists of genes to could be excluded or should be carefully considered for specific contrasts. This file contains a contrast field and the corresponding genes written as gene symbols.
+>    - **Pseudobulk Plot:** A visual representation of the pseudobulk data.
+>    ![Pseudobulk Plot Example](../../images/pseudobulk-analysis/Decoupler_Pseudobulk_Plot.png)
+>    - **Filter by Expression Plot:** Illustrates the expression filtering applied to the data.
+>    ![Filter by Expression Plot Example](../../images/pseudobulk-analysis/Decoupler_Pseudobulk_Plot.png)
+>      **Pseudobulk AnnData file:**: An AnnData file that contains the aggregated pseudobulks.  
+>
+> 3. The **pseudobulk count matrix** is the primary input required for analysis using **edgeR**, a tool designed for differential expression analysis. The **Samples Metadata** is another file that will serve as an input for the edgeR tool. 
+>
+{: .solution}
+
+
+## Sanitation Steps - Part 1
+
+The next steps will help you refine your data for easier handling. We will use standard galaxy tools, like: [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1), [Remove columns](toolshed.g2.bx.psu.edu/repos/iuc/column_remove_by_header/column_remove_by_header/1.0) and [Text reformatting](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1). 
```

Comment:
for regular links to the toolshed you should include the https:// part of the url. Alternatively you can use the {% tool []() %} syntax like you do in the hands-on boxes